testing/test.go

```diff
@@ -26,9 +26,65 @@ import (
 // NewTestLogger returns a logr.Logger that prints through a testing.T object.
 // Info logs are only enabled at V(0).
 func NewTestLogger(t *testing.T) logr.Logger {
-	fn := func(prefix, args string) {
-		t.Helper()
-		t.Logf("%s: %s", prefix, args)
+	l := &testlogger{
+		Formatter: funcr.NewFormatter(funcr.Options{}),
```

Comment:
I think it would make sense to allow configuring the formatting, in particular the log level threshold. I've not added it to the PR to not bloat it further, but would like to see this addressed before logr v1.1.0.

The simplest solution would be:
```
func NewTestLoggerWithOptions(t *testing.T, opts funcr.Options) logr.Logger {
   ...
}
```

Reply:
I'd rather see TestLogger get its own options struct, even if that is just a typedef to funcr.Options.